pkg/ingester/user_tsdb_test.go

```diff
@@ -103,6 +104,40 @@ func TestUserTSDB_acquireAppendLock(t *testing.T) {
 	})
 }
 
+func TestUserTSDB_blocksToDelete(t *testing.T) {
+	t.Run("blocksToDelete is concurrency safe and is non-racy", func(*testing.T) {
+		var wg sync.WaitGroup
+		wg.Add(2)
+
+		setDBStarted := make(chan struct{})
+		blockDeleteStarted := make(chan struct{})
+		var userDB userTSDB
+
+		go func() {
+			defer wg.Done()
+
+			<-setDBStarted
+
+			for i := 0; i < 1000; i++ {
+				userDB.blocksToDelete(nil)
+				if i == 0 {
+					close(blockDeleteStarted)
+				}
+			}
+		}()
+
+		go func() {
+			defer wg.Done()
+
+			close(setDBStarted)
+			<-blockDeleteStarted
+			userDB.setDB(&tsdb.DB{})
+		}()
+
+		wg.Wait()
+	})
+}
```

Comment:
I don't think this test is useful. It's 30+ lines to test that we've actually set the bool, but it doesn't even check that we're calling `userDB.setDB(db)` instead of just doing `userDB.db = db`

Reply:
For reference in case it wasn't clear, the test is for catching data races. I'll have to rethink it though, after applying your suggestion above.